src/ShallowWaterIMEX.jl

```diff
@@ -0,0 +1,216 @@
+function shallow_water_imex_time_step!(
+     h₂, u₂, uₚ, ϕ, F, q₁, q₂,                               # in/out args
+     H1h, H1hchol, h_wrk, u_wrk, A_wrk, Bchol,               # more in/out args
+     model, dΩ, dω, V, P, Q, R, S, f, g, h₁, u₁, u₀,         # in args
+     RTMMchol, L2MMchol, RTMM, L2MMinvD, dt, τ, leap_frog)   # more in args
+
+  # energetically balanced implicit-explicit second order shallow water solver
+  # reference: eqns (31-33) of
+  # https://github.com/BOM-Monash-Collaborations/articles/blob/main/energetically_balanced_time_integration/EnergeticallyBalancedTimeIntegration_SW.tex
+  #
```

Comment:
The signature description is outdated. I think the same applies to the other functions. I would either update (if we think they are strictly necessary) or remove them (if we think that their names are sufficiently meaning revealing).

Reply:
Good point, I've removed them.